Objects/tupleobject.c

```diff
@@ -568,7 +569,15 @@ tuple_index_impl(PyTupleObject *self, PyObject *value, Py_ssize_t start,
         else if (cmp < 0)
             return NULL;
     }
-    PyErr_SetString(PyExc_ValueError, "tuple.index(x): x not in tuple");
+
+    if (!(value_repr = PyObject_Repr(value))) {
```

Comment:
Please don't do assignment in an `if` guard as it's hard to notice it's not a comparison.